packages/tools/fetch-tool/src/fluidFetchSnapshot.ts

```diff
@@ -76,7 +77,7 @@ function fetchBlobs(prefix: string,
                 reused = false;
                 blob = blobCache.get(blobId);
                 if (blob === undefined) {
-                    blob = storage.read(blobId);
+                    blob = bufferToString(await storage.readBlob(blobId));
```

Comment:
Note that you changes the behavior of the function.
It used to be that this code would issue hundreds or parallel requests to read, and they all would run in parallel.
Now, with await here, we wait for one read to finish before issuing another read. This will substantially slow down fetch tool.
Your intention to simplify code is great, but we need to balance it with perf considerations. I.e. either return it to work with promises, or find a way for this function to do parallel reads, but fill in buffers, not promise to buffers.
The later can likely be done by changing for loop on line 70 to something like

await Promise.all(Object.keys(tree.blobs),map(async (key) => {...}));

Then each iteration is its own independent async function, and await in it will unlock next iteration, not block it.